### PR TITLE
ART-9190 Build Plashets only when required

### PR DESCRIPTION
**Pull request by thegreyd**
https://issues.redhat.com/browse/ART-9190

Skip plashet sync if there have been no changes to rpms since latest brew event stored in plashet.yml
https://ocp-artifacts.hosts.prod.psi.rdu2.redhat.com/pub/RHOCP/plashets/4.16/stream/el8/latest/plashet.yml

Test run: https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Focp4/159/console

### CODE REVIEW

**Review by vfreex — COMMENTED** [review]
/lgtm
/approve

File: doozer/doozerlib/cli/config_plashet.py [hunk 1]
```diff
@@ -535,7 +538,12 @@ def from_tags(config: SimpleNamespace, brew_tag: Tuple[Tuple[str, str], ...], em
             assembly = 'stream'
 
         # If assemblies are disabled, the true latest rpm builds from the tag will be collected; Otherwise we will only collect the rpm builds specific to that assembly.
-        tagged_builds = builder.from_tag(tag, inherit, assembly, event)
+        tagged_builds = builder.from_tag(tag, inherit, assembly, event=event, after_event=after_brew_event)
+        if not tagged_builds:
+            logger.info(f"Tag {tag} has no change for tagged rpms since --after-brew-event={after_brew_event}. "
+                        "Skipping plashet generation")
+            exit(0)
```
Comment (vfreex) [inline doozer/doozerlib/cli/config_plashet.py:545]: I think we probably need a special exit code to signal the caller (pyartcd) not to update symlinks

Comment (thegreyd) [reply doozer/doozerlib/cli/config_plashet.py:545]: Good catch! I'll update it

Comment (thegreyd) [reply doozer/doozerlib/cli/config_plashet.py]: https://github.com/openshift-eng/art-tools/pull/476/commits/b3f312f2c09a4a90de567c6800e0684a3d10ec36

**Comment by vfreex** [comment]
I found another issue: It will completely ignore assembly overrides if no tagging changes.

**Comment by thegreyd** [comment]
Thanks @vfreex Yes that was something I chose to ignore when I first wrote this, since wanted to get something out quickly to help with our plashet building issues. Now that those issues are resolved, this PR isn't in a rush to get in and we can think more closely about assembly/group overrides.

I wrote https://issues.redhat.com/browse/ART-9190 to capture this, please see and add if you have anything.

For assembly/group overrides I'm thinking of computing a digest of those and recording that into plashet.yml as well

**Pull request closed without merging**